solarforecastarbiter/io/fetch/nwp.py

```diff
@@ -0,0 +1,669 @@
+            for hr in range(0, 24, int(model['update_freq'].strip('h'))):
+                hrpath = path / f'{hr:02d}'
+                glob = list(hrpath.glob('*.nc'))
+                if len(glob) == 0:
+                    no_file.append(pd.Timestamp(f'{dir_[:8]}T{hr:02d}00Z'))
+        else:
+            hrpath = model_path / dir_[:4] / dir_[4:6] / dir_[6:8] / dir_[8:10]
+            glob = list(hrpath.glob('*.nc'))
+            if len(glob) == 0:
+                no_file.append(pd.Timestamp(f'{dir_[:8]}T{dir_[8:10]}00Z'))
+    if len(no_file) == 0:
+        # all files for current dirs present, get next day
+        inittime = max(
+            [pd.Timestamp(f'{dir_[:8]}T0000Z')]) + pd.Timedelta('1d')
+    else:
+        inittime = min(no_file)
+    await sleep_until_inittime(inittime, model)
+    return inittime
+
+
+async def next_run_time(inittime, modelpath, model):
+    inittime += pd.Timedelta(model['update_freq'])
+    # check if nc file exists for this inittime
+    if len(list(
+            (modelpath / inittime.strftime('%Y/%m/%d/%H')).glob('*.nc'))) > 0:
+        return await next_run_time(inittime, modelpath, model)
+    await sleep_until_inittime(inittime, model)
+    return inittime
+
+
+async def run(basepath, model_name, chunksize, once=False):
+    session = make_session()
+    modelpath = basepath / model_name
+    model = model_map[model_name]
+    inittime = await startup_find_next_runtime(modelpath, session, model)
+    while True:
+        fetch_tasks = set()
+        async for params in files_to_retrieve(session, model, inittime):
+            fetch_tasks.add(asyncio.create_task(
+                fetch_grib_files(session, params, modelpath, inittime,
+                                 chunksize)))
+        files = await asyncio.gather(*fetch_tasks)
+        if len(files) != 0:  # skip to next inittime
+            path_to_files = files[0].parent
+            nctmpfile = await process_grib_to_netcdf(path_to_files)
+            try:
+                await optimize_netcdf(
+                    nctmpfile, path_to_files / f'{model_name}.nc')
+            except Exception:
+                raise
+            else:
+                # remove grib files
+                for f in files:
+                    f.unlink()
+        if once:
+            break
+        else:
+            logger.info('Moving on to next model run')
+            inittime = await next_run_time(inittime, modelpath, model)
+    await session.close()
+
+
+async def optimize_only(path_to_files, model_name):
+    nctmpfile = await process_grib_to_netcdf(path_to_files)
+    try:
+        await optimize_netcdf(
+            nctmpfile, path_to_files / f'{model_name}.nc')
+    except Exception:
+        raise
+    else:
+        # remove grib files
+        for f in path_to_files.glob('*.grib2'):
+            f.unlink()
+
+
+def check_wgrib2():
+    if shutil.which('wgrib2') is None:
+        logger.error('wgrib2 was not found in PATH and is required')
```

Comment:
is this pulled in through the python dependencies? If not, `pip install .[fetch]` should fail if it's not available.

Reply:
Though we can leave that for a future issue, too.

Reply:
future #64